walletlibrary/src/test/java/com/microsoft/walletlibrary/serializer/PresentationExchangeResponseBuilderTest.kt

```diff
@@ -0,0 +1,122 @@
+import com.microsoft.walletlibrary.util.LibraryConfiguration
+import com.microsoft.walletlibrary.verifiedid.StringVCSerializer
+import com.microsoft.walletlibrary.verifiedid.VerifiableCredential
+import io.mockk.coEvery
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.slot
+import io.mockk.spyk
+import io.mockk.verify
+import kotlinx.coroutines.runBlocking
+import kotlinx.serialization.json.Json
+import java.util.Date
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+
+
+internal class PresentationExchangeResponseBuilderTest {
+    val signer = mockk<TokenSigner>()
+    val payload = slot<String>()
+    val subject = slot<Identifier>()
+    val serializer = Json
+    val libraryConfiguration = mockk<LibraryConfiguration>()
+    val signedPayload = "expectedSignature${(0..20).random()}"
+
+    fun makeVerifiedId (sub: String): VerifiableCredential {
+        return VerifiableCredential(
+            com.microsoft.walletlibrary.did.sdk.credential.models.VerifiableCredential(
+                "",
+                "",
+                VerifiableCredentialContent(
+                    "",
+                    VerifiableCredentialDescriptor(emptyList(), listOf("TestCredential"), emptyMap()),
+                    sub,
+                    sub,
+                    Date().time
+                )
+            ), VerifiableCredentialContract(
+                sub,
+                InputContract(
+                    credentialIssuer = sub,
+                    issuer = sub,
+                ),
+                DisplayContract(
+                    card = CardDescriptor("", "", "", "", null, ""),
+                    consent = ConsentDescriptor(instructions = ""),
+                    claims = emptyMap()
+                )
+            )
+        )
+    }
+
+    val identifier: Identifier = Identifier(
+        "did:example:mockabcdefghi1234567890",
+        "sig",
+        "enc",
+        "sig",
+        "sig",
+        "mock"
+    )
+    val correctVerifiedId = makeVerifiedId(identifier.id)
+    val defaultRequirement = PresentationExchangeVerifiedIdRequirement(
+        id = "recOne",
+        types = listOf("TestCredential"),
+        inputDescriptorId = "1"
+    )
+
+    init {
+        val mockIdentifierManager = mockk<IdentifierManager>()
+        coEvery { mockIdentifierManager.getMasterIdentifier() } returns Result.Success(identifier)
+        every { libraryConfiguration.serializer } returns serializer
+        every { libraryConfiguration.tokenSigner } returns signer
+        every { libraryConfiguration.identifierManager } returns mockIdentifierManager
+        every { signer.signWithIdentifier( capture(payload), capture(subject) )} returns signedPayload
+        assertTrue(defaultRequirement.fulfill(correctVerifiedId).isSuccess)
+    }
+
+    @Test
+    fun serialize_singleRequirement_traversesRequirement() {
```

Comment:
Could you add a couple of tests for buildIdToken and buildVpToken as well?

Reply:
Absolutely! I just realized this has a few other changes to resolve issues with integration and didn't want to hold it up over additional test cases.